src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerForTag.java

```diff
@@ -86,9 +86,10 @@ public String eagerInterpret(
           interpreter,
           true
         );
+
+      // Run for loop again now that the necessary values have been deferred
+      eagerExecutionResult = runLoopOnce(tagNode, interpreter);
```

Comment:
Moved this because we should only have to run the for loop again if there are some speculative bindings. It's just an optimization, and it's unrelated to the reset of the changes here. (The logic for re-running a deferred for loop is very similar to how we need to rerun the deferred macro so that any values that got deferred the first time start off being deferred. This is because a for loop is much like a recursive macro function)